Snake/BusinessLayer/Engine.cs

```diff
@@ -1,217 +1,217 @@
+                    mazeArray[head.returnX(), head.returnY()] = snakeHead;
+
+                    if ((newSnakeHead.returnX() == food.xLocation) && (newSnakeHead.returnY() == food.yLocation))
+                    {
+                        Score.incrementScore(food.pointsWorth);
+
+                    }
+
+                    food = null;
+                    bool isValid = true;
+                    do
+                    {
+                        food = foodGenerator.generateFood(mazeLength, mazeWidth);
+                        isValid = validateNewFoodLocation(food);
+                    } while (!isValid);
+
+                    mazeArray[food.xLocation, food.yLocation] = Food;
+                    break;
+
+                default:   // snake moves
+                    List<Point> SnakeCurrentPosition = gameSnake1.returnCurrentSnakePosition();
+                    mazeArray[SnakeCurrentPosition.Last().returnX(), SnakeCurrentPosition.Last().returnY()] = (int)Elements.blank;
+                    mazeArray[SnakeCurrentPosition.First().returnX(), SnakeCurrentPosition.First().returnY()] = (int)Elements.snakeBody;
+                    snakesNewLocation = gameSnake1.snakeMove(snakeDirection, false);
+                    mazeArray[snakesNewLocation.First().returnX(), snakesNewLocation.First().returnY()] = (int)Elements.snakeHead;
+
+                    mazeArray[food.xLocation, food.yLocation] = Food;
+                    break;
+            }       
+            return mazeArray;
+        }
+
+        private Point getNewHead(Direction snakeDirection)
+        {
+            List<Point> snakeBody = gameSnake1.returnCurrentSnakePosition();
+
+            // Check current snake head location.
+            Point snakeHead = snakeBody.First();
+
+            // Cross check with new snake head location.
+            Point newSnakeHead;
+            int x = snakeHead.returnX();
+            int y = snakeHead.returnY();
+            switch (snakeDirection)
+            {
+
+                case Direction.Right:
+                    newSnakeHead = new Point(x, y + step);
+                    break;
+
+                case Direction.Left:
+                    newSnakeHead = new Point(x, y - step);
+                    break;
+
+                case Direction.Up:
+                    newSnakeHead = new Point(x - step, y);
+                    break;
+
+                case Direction.Down:
+                    newSnakeHead = new Point(x + step, y);
+                    break;
+
+                default:
+                    throw new System.Exception("Invalid direction.");
+            }
+            return newSnakeHead;
+        }
+        public bool validateNewFoodLocation(Food newFood)
+        {
+            int x = newFood.xLocation;
+            int y = newFood.yLocation;
+
+            if ((x >= mazeLength) || (y >= mazeWidth))
+            {
+                return false;
+            }
+            if (mazeArray[x, y] == (int)Elements.mazeBody)
```

Comment:
seems like your maze array should not be an array of ints but instead an array of enum types. then you would not have to keep casting